### PR TITLE
ACS-6380: [HxI LI] Added Camel-retry logic to PreSignedUrlRequester

### PR DESCRIPTION
**Pull request by krdabrowski**
[HxI LI] Upload pdf to S3 - PART II - added Camel-retry logic to PreSignedUrlRequester
https://alfresco.atlassian.net/browse/ACS-6380

### CODE REVIEW

File: live-ingester/src/main/resources/application.yml [hunk 1]
```diff
@@ -22,6 +22,19 @@ alfresco:
...
+        deleyMultiplier: 2
+        reasons:
+          - org.alfresco.hxi_connector.live_ingester.domain.exception.EndpointServerErrorException
+          - java.net.UnknownHostException
+          - java.net.MalformedURLException
+          - com.fasterxml.jackson.core.io.JsonEOFException
+          - com.fasterxml.jackson.databind.exc.MismatchedInputException
+          - org.apache.hc.client5.http.HttpHostConnectException
+          - org.apache.hc.core5.http.MalformedChunkCodingException
+          - org.apache.hc.core5.http.NoHttpResponseException
```
Comment (tpage-alfresco) [inline live-ingester/src/main/resources/application.yml:37]: This is clever, but I'm not sure I'd want a customer to have to (or even be able to) configure these.

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/storage/connector/PreSignedUrlRequester.java [hunk 2]
```diff
@@ -29,64 +29,81 @@
...
+                .logExhaustedMessageBody(true)
+                .log(LoggingLevel.ERROR, log, "Unexpected response. Headers: ${headers}, Body: ${body}")
+                .maximumRedeliveries(integrationProperties.storage().retry().attempts())
+                .redeliveryDelay(integrationProperties.storage().retry().initialDelay())
+                .backOffMultiplier(integrationProperties.storage().retry().delayMultiplier())
                 .stop();
+        }
+
+        onException(Exception.class)
+            .log(LoggingLevel.ERROR, log, "Unexpected response. Headers: ${headers}, Body: ${body}")
```
Comment (tpage-alfresco) [inline live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/storage/connector/PreSignedUrlRequester.java:84]: Maybe make this log message different from the one above? e.g. 
```suggestion
            .log(LoggingLevel.ERROR, log, "Unexpected unretryable response. Headers: ${headers}, Body: ${body}")
```

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/storage/connector/PreSignedUrlRequester.java [hunk 3]
```diff
@@ -29,64 +29,81 @@
...
+    @SuppressWarnings("unchecked")
     public void configure()
     {
-        onException(Exception.class)
-                .log(LoggingLevel.ERROR, log, "Unexpected response. Body: ${body}")
+        List<Class<? extends Throwable>> retryReasons = integrationProperties.storage().retry().reasons();
+        // @formatter:off
+        if (retryReasons != null && !retryReasons.isEmpty())
+        {
+            onException(retryReasons.toArray(Class[]::new))
```
Comment (tpage-alfresco) [inline live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/storage/connector/PreSignedUrlRequester.java:73]: Presumably this applies to both routes defined below, in which case what is the point of having an intermediate `LOCAL_RETRYABLE_ENDPOINT`?

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/storage/connector/PreSignedUrlRequester.java [hunk 4]
```diff
@@ -29,64 +29,81 @@
...
 
     @Override
+    @SuppressWarnings("unchecked")
     public void configure()
     {
-        onException(Exception.class)
-                .log(LoggingLevel.ERROR, log, "Unexpected response. Body: ${body}")
+        List<Class<? extends Throwable>> retryReasons = integrationProperties.storage().retry().reasons();
+        // @formatter:off
+        if (retryReasons != null && !retryReasons.isEmpty())
```
Comment (Pawel-608) [inline live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/storage/connector/PreSignedUrlRequester.java:71]: Could we use `@NotEmpty` on reasons property instead?

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/storage/connector/PreSignedUrlRequester.java (hunk 2 above)
Comment (Pawel-608) [reply live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/storage/connector/PreSignedUrlRequester.java:84]: I agree, but I have one more question: this message will be logged if error won't be catched by `onException(retryReasons.toArray(Class[]::new))` right?

File: live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/storage/connector/PreSignedUrlRequester.java (hunk 3 above)
Comment (Pawel-608) [reply live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/storage/connector/PreSignedUrlRequester.java:73]: +1

**Pull request closed without merging**